protocols/noise/tests/smoke.rs

```diff
@@ -30,65 +34,70 @@ use tokio::{self, io};
 fn xx() {
     let _ = env_logger::try_init();
     fn prop(message: Vec<u8>) -> bool {
+        let server_keypair = Keypair::new();
+        let server_transport = TcpConfig::new().with_upgrade(XX::new(server_keypair));
 
-        let server_keypair = Keypair::<X25519>::new();
-        let server_transport = TcpConfig::new().with_upgrade(NoiseConfig::xx(server_keypair));
-
-        let client_keypair = Keypair::<X25519>::new();
-        let client_transport = TcpConfig::new().with_upgrade(NoiseConfig::xx(client_keypair));
+        let client_keypair = Keypair::new();
+        let client_transport = TcpConfig::new().with_upgrade(XX::new(client_keypair));
 
         run(server_transport, client_transport, message);
         true
     }
-    QuickCheck::new().max_tests(30).quickcheck(prop as fn(Vec<u8>) -> bool)
+    QuickCheck::new()
+        .max_tests(30)
+        .quickcheck(prop as fn(Vec<u8>) -> bool)
 }
 
 #[test]
 fn ix() {
     let _ = env_logger::try_init();
     fn prop(message: Vec<u8>) -> bool {
+        let server_keypair = Keypair::new();
+        let server_transport = TcpConfig::new().with_upgrade(IX::new(server_keypair));
 
-        let server_keypair = Keypair::<X25519>::new();
-        let server_transport = TcpConfig::new().with_upgrade(NoiseConfig::ix(server_keypair));
-
-        let client_keypair = Keypair::<X25519>::new();
-        let client_transport = TcpConfig::new().with_upgrade(NoiseConfig::ix(client_keypair));
+        let client_keypair = Keypair::new();
+        let client_transport = TcpConfig::new().with_upgrade(IX::new(client_keypair));
 
         run(server_transport, client_transport, message);
         true
     }
-    QuickCheck::new().max_tests(30).quickcheck(prop as fn(Vec<u8>) -> bool)
+    QuickCheck::new()
+        .max_tests(30)
+        .quickcheck(prop as fn(Vec<u8>) -> bool)
 }
 
 #[test]
 fn ik_xx() {
     let _ = env_logger::try_init();
     fn prop(message: Vec<u8>) -> bool {
-        let server_keypair = Keypair::<X25519>::new();
-        let server_public = server_keypair.public().clone();
-        let server_transport = TcpConfig::new()
-            .and_then(move |output, endpoint| {
-                if endpoint.is_listener() {
-                    Either::A(apply_inbound(output, NoiseConfig::ik_listener(server_keypair)))
-                } else {
-                    Either::B(apply_outbound(output, NoiseConfig::xx(server_keypair)))
-                }
-            });
-
-        let client_keypair = Keypair::<X25519>::new();
-        let client_transport = TcpConfig::new()
-            .and_then(move |output, endpoint| {
-                if endpoint.is_dialer() {
-                    Either::A(apply_outbound(output, NoiseConfig::ik_dialer(client_keypair, server_public)))
-                } else {
-                    Either::B(apply_inbound(output, NoiseConfig::xx(client_keypair)))
-                }
-            });
+        let server_keypair = Keypair::new();
```

Comment:
Nit: while I don't have a strong opinion on style, it's always a bit annoying when a large portion of the diff consists in insubstantial style changes.